### PR TITLE
Advanced Plastic Surgery 2.0, It's rare now.

### PR DESCRIPTION
**Pull request by Singul0**
## About The Pull Request
A revival of #76508 which got closed due to it eating up genetics niche. This PR fixess that issue by seperating the surgery into two. the normal roundstart plastic surgery and advanced plastic surgery. both works identically with the one exception being that the advanced one allows the feature that the last PR tried to add. which is to allow surgery to mimic people in photographs.
the regular one remains as-is.

The disk containing the surgery can be found in a lot of places, a role-restricted item to doctors and roboticists for 1TC, as a rare maint loot and BEPIS technode reward
## Why It's Good For The Game
Opportunity for more fun gimmicks, ammo for BB's and paranoia, staple of SS13 gameplay.
This also retroactively buffs forensic scanners and records since it really only changes your name and voice. You can change your hair all you want but anything more than that is a trip to genetics. overall allows for more engaging gameplay while not detracting from genetics niche  with how niche and rare it would appear in rounds.
## Changelog
:cl:
add: Adds an advanced plastic surgery procedure, allowing you to imitate people in pictures. Simply hold a picture in your offhand of the person you wish to imitate as while conducting the surgery! Remember, it's not foolproof, it only changes your name and voice!
adds: You can obtain the disk containing the afromentioned surgery. as a role-restricted item to doctors and roboticists for 1TC, as a rare maint loot and BEPIS technode reward 
/:cl:

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
Respectfully, how does locking this behind a TC wall or a research/money wall contend with why I closed the PR initially? Even if you overlooked the paragraphs of stepping on genetics toes and looked strictly about an off comment regarding being able to do it roundstart, thats still possible with the TC method or just getting lucky and finding it in maint.

Did I say this was ok in discord or something? Like Im trying to read the last PR and Im not understanding how you came to the conclusion that the PR was ok if you just gave it to doctors (not even genetics) if you tweaked the accessibility slightly for the people who would be the prime users of this feature.

**Comment by Time-Green** [comment]
> Respectfully, how does locking this behind a TC wall or a research/money wall contend with why I closed the PR initially? Even if you overlooked the paragraphs of stepping on genetics toes and looked strictly about an off comment regarding being able to do it roundstart, thats still possible with the TC method or just getting lucky and finding it in maint.
> 
> Did I say this was ok in discord or something? Like Im trying to read the last PR and Im not understanding how you came to the conclusion that the PR was ok if you just gave it to doctors (not even genetics) if you tweaked the accessibility slightly for the people who would be the prime users of this feature.

I’m not really sure what the big deal is? I agree it’s important to stop feature creep, but this is not really a major mechanic. The plastic surgery doesn’t need consent from the person you’re copying (people don’t just walk into scanners, and corpses dont work), but doesn’t change features. If an assistant finds a main surgery disk, photograps the captain and gets plastic surgery, then that’s fun good RP

**Comment by Singul0** [comment]
> How does locking this behind a TC wall or a research/money wall contend with why I closed the PR initially? Even if you overlooked the paragraphs of stepping on genetics toes

By making it more rare, This wouldn't be stepping on genetics little niche, You'd really only see it in once in a blue moon round where someone actually does BEPIS and for it to unlock the right technode or to find it in the rare (and really dilluted) maint loot  list when out scavenging in maintenance.

Your first thought when you need to disguise as someone wouldn't be to go to medical every single round. But to genetics where you could far more easily and comprehensively disguise as someone since you could copy someone genetic makeup in seconds using the sequencer.

If anything, this also buffs genetics unintentionally, Since more people would be on the lookout for the signs of an imperfect disguise, having a complete genetic makeup disguise gives you far more credibliity..

Honestly, I was just bored and wanted something to work on.

**Comment by moocowswag** [comment]
Overlapping abilities between jobs is not stepping on each others toes though, xenobio , botany, and chemistry, can all make healing chems, roboticists can do surgeries that medical staff can do. Geneticists can make genes that buff the player in seemingly unique ways but alot of them are achievable from other jobs, fiery sweats 50% burn resistance is functionally achieveable with appropriate armor values or slime person,

 I know its a stretch to say hulk is technically in comparison to xenobio spamming sentient mobs but its true, sentient mobs are also stun immune and cant be disarmed and can break down walls/doors infinitely while hulk can use items and benefit from chems, while xenobio mobs are usually not the player spamming them and instead are making them without downsides to their actual character.

Also this....plastic surgery 2.0.....the ability to change your identity....everything....

you can already do that with a renaming potion from xenobio and pick the exact name you want to change to....hows that for stepping on geneticist`s toes.

File: code/modules/surgery/plastic_surgery.dm [hunk 1]
```diff
@@ -43,8 +83,15 @@
...
 		if(!isabductor(user))
-			for(var/i in 1 to 10)
-				names += target.dna.species.random_name(target.gender, TRUE)
+			var/obj/item/offhand = user.get_inactive_held_item()
+			if(istype(offhand, /obj/item/photo) && istype(surgery, /datum/surgery/plastic_surgery/advanced))
+				var/obj/item/photo/disguises = offhand
+				for(var/namelist as anything in disguises.picture?.names_seen)
+					names += namelist
+			else
+				user.visible_message(span_warning("You have no picture to base the appearance on, Reverting to random appearances."))
```
Comment (Time-Green) [inline code/modules/surgery/plastic_surgery.dm:92]:
```suggestion
				user.visible_message(span_warning("You have no picture to base the appearance on, reverting to random appearances."))
```

File: code/modules/surgery/plastic_surgery.dm [hunk 2]
```diff
@@ -1,3 +1,8 @@
+/obj/item/disk/surgery/advanced_plastic_surgery
```
Comment (Time-Green) [inline code/modules/surgery/plastic_surgery.dm:1]: I'd appreciate it if you autodocced the type

File: code/modules/surgery/plastic_surgery.dm [hunk 3]
```diff
@@ -9,6 +14,41 @@
...
+	name = "advanced plastic surgery"
+	steps = list(
+		/datum/surgery_step/incise,
+		/datum/surgery_step/retract_skin,
+		/datum/surgery_step/insert_plastic,
+		/datum/surgery_step/reshape_face,
+		/datum/surgery_step/close,
+	)
+
+//insert plastic
```
Comment (Time-Green) [inline code/modules/surgery/plastic_surgery.dm:27]: :(

File: code/modules/uplink/uplink_items/job.dm [hunk 4]
```diff
@@ -173,6 +173,16 @@
 	cost = 5
 	surplus = 50
 
+/datum/uplink_item/role_restricted/advanced_plastic_surgery
+	name = "Advanced Plastic Surgery Program"
+	desc = "A bootleg copy of an collector item, this disk contains the procedure to perform advanced plastic surgery, allowing you to model someone's face and voice based on a picture taken by a camera on your offhand. \
+	All changes are superficial and does not change ones genetic makeup \
```
Comment (Time-Green) [inline code/modules/uplink/uplink_items/job.dm:179]: Punctuation?

**Review by Time-Green — CHANGES_REQUESTED** [review]

File: code/modules/surgery/plastic_surgery.dm (hunk 3 above)
Comment (Singul0) [reply code/modules/surgery/plastic_surgery.dm:27]: I have zero idea what you mean :(
I added a happy face in the comment to compensate :)

**Comment by Singul0** [comment]
God bless you 🙏 🙏 🙏

File: code/modules/surgery/plastic_surgery.dm (hunk 3 above)
Comment (Time-Green) [reply code/modules/surgery/plastic_surgery.dm]: Aight fair enough

**Pull request merged**